Add ChartDashboard with Charts, Export Functionality, and Test Coverage

Rajeshgulivindala opened the pull request:
This PR implements the Enrollment Dashboard as part of Objective3. It provides instructors with a visual summary of student enrollment data across GitHub and Microsoft Loop.

## Features Implemented:
Bar, Line, and Pie charts using Recharts to display enrollment trends and status distribution

Export dashboard view as PNG using html-to-image and file-saver

Toggle view with "Back to Main" button

Responsive layout using Tailwind CSS

## Test Coverage:

Added Jest test file EnrollmentDashboard.test.tsx

Verified rendering of:

 1. Dashboard heading and section titles

 2. All three chart types

 3. Table breakdown

 4. Interactive buttons

achow-conestogac commented on src/index.css:
this file does not seem like it should be done in this PR.

achow-conestogac commented on src/EnrollmentDashboard.test.tsx:
```diff
@@ -0,0 +1,47 @@
+  disconnect() {}
+};
+
+jest.mock('html-to-image', () => ({
+  toPng: jest.fn(() => Promise.resolve('data:image/png;base64,mocked')),
+}));
+jest.mock('file-saver', () => ({
+  saveAs: jest.fn(),
+}));
```
Please do not use jest.mock.

define an interface in order to create tests

achow-conestogac reviewed (commented):
I'm pausing the review here as there's so many overlaps between this PR and [this PR](https://github.com/conestogac-acsit/SENG8130-Software-Quality-Applications-Lab/pull/230)

There's definitely some communication/integration issues. 

Until this is resolved, I cannot proceed to review.

Rajeshgulivindala commented:
After Discussion with Nirajan we removed the EnrollmentDashboard Button. Now there is no overlapping between us for integration

As you mentioned ,
"this file does not seem like it should be done in this PR"
we have removed this file

''Please do not use jest.mock.
define an interface in order to create tests''

 jest.mock, we have removed it and added the interface in it

achow-conestogac commented:
> After Discussion with Nirajan we removed the EnrollmentDashboard Button. Now there is no overlapping between us for integration
> 
> As you mentioned , "this file does not seem like it should be done in this PR" we have removed this file
> 
> ''Please do not use jest.mock. define an interface in order to create tests''
> 
> jest.mock, we have removed it and added the interface in it

I disagree.  EnrollmentDashboard is still in both your PRs. This is the same for `Student` interface definitions.
